src/read/rnglists.rs

```diff
@@ -491,32 +504,41 @@ impl<R: Reader> RngListIter<R> {
 
             let range = match raw_range {
                 RawRngListEntry::BaseAddress { addr } => {
-                    self.base_address = addr;
+                    self.base_address = try_tombstone!(addr);
                     continue;
                 }
                 RawRngListEntry::BaseAddressx { addr } => {
-                    self.base_address = self.get_address(addr)?;
+                    self.base_address = try_tombstone!(self.get_address(addr)?);
                     continue;
                 }
                 RawRngListEntry::StartxEndx { begin, end } => {
-                    let begin = self.get_address(begin)?;
-                    let end = self.get_address(end)?;
+                    let begin = try_tombstone!(self.get_address(begin)?);
+                    let end = try_tombstone!(self.get_address(end)?);
```

Comment:
Instead of checking this for each kind of entry, do a single check below at the same place we check `if range.begin > range.end`.